R/add_sda_target.R

```diff
@@ -0,0 +1,185 @@
+#' Add portfolio level SDA target
+#'
+#' This function calculates the portfolio-level emission factor targets, as
+#' calculated using the sectoral-decarbonisation approach.
+#'
+#' @param data A dataframe like the output of
+#'   [r2dii.match::prioritize()].
+#' @param ald An asset level dataframe like [r2dii.data::ald_demo].
+#' @param co2_intensity_scenario A scenario dataframe like
+#'   [r2dii.analysis::co2_intensity_scenario].
+#'
+#' @return  A tibble with portfolio-level emission_factor targets.
+#' @export
+#'
+#' @examples
+#' installed <- requireNamespace("r2dii.data", quietly = TRUE) &&
+#'   requireNamespace("r2dii.match", quietly = TRUE)
+#' if (!installed) stop("Please install r2dii.match and r2dii.data")
+#'
+#' library(r2dii.data)
+#' library(r2dii.match)
+#'
+#' valid_matches <- match_name(loanbook_demo, ald_demo) %>%
+#'   # WARNING: Remember to validate matches (see `?prioritize`)
+#'   prioritize()
+#'
+#' valid_matches %>%
+#'   add_sda_target(
+#'     ald = ald_demo,
+#'     co2_intensity_scenario = r2dii.analysis::co2_intensity_scenario
+#'   )
+add_sda_target <- function(data, ald, co2_intensity_scenario) {
+  start_year <- co2_intensity_scenario %>%
+    select(.data$sector, .data$year) %>%
+    group_by(.data$sector) %>%
+    summarize(start_year = min(.data$year))
```

Comment:
Every time we call `summarize()` with a function `f()` we need to think and test how `f()` handles `NA`. For example, `min()`  defaults to `na.rm = TRUE`; better to be explicit to about this -- it may cause errors that we need to test for.